src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -737,6 +738,52 @@ PhasePartition(ts::PhaseDry{FT}) where {FT<:Real} = PhasePartition(FT(0), FT(0),
 PhasePartition(ts::PhaseEquil) = PhasePartition_equil(air_temperature(ts), air_density(ts), ts.q_tot)
 PhasePartition(ts::PhaseNonEquil) = ts.q
 
+function roots_∂e_int_∂T(T::FT, e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  cvm = cv_m(PhasePartition(q_tot))
+  q_vap_sat = q_vap_saturation(T, ρ)
+  λ = liquid_fraction_equil(T)
+  L = λ*FT(LH_v0) + (1-λ)*FT(LH_s0)
+  ∂q_vap_sat_∂T = q_vap_sat*L/(FT(R_v)*T^2)
+  # T0 = FT(T_min)
+  T0 = FT(T_0)
+  return cvm + ( FT(e_int_v0) + (1-λ)*FT(e_int_i0) + (T - T0) )*∂q_vap_sat_∂T
+end
+
+"""
+    saturation_adjustment_NewtonsMethod(e_int, ρ, q_tot)
+
+Compute the temperature that is consistent with
+
+ - `e_int` internal energy
+ - `ρ` (moist-)air density
+ - `q_tot` total specific humidity
+
+using Newtons method with analytic gradients.
+
+See also [`saturation_adjustment`](@ref).
+"""
+function saturation_adjustment_NewtonsMethod(e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  T_1 = max(FT(T_min), air_temperature(e_int, PhasePartition(q_tot))) # Assume all vapor
+  q_v_sat = q_vap_saturation(T_1, ρ)
+  unsaturated = q_tot <= q_v_sat
+  if unsaturated
+    return T_1
+  else
+    # FIXME here: need to revisit bounds for saturation adjustment to guarantee bracketing of zero.
+    T_2 = air_temperature(e_int, PhasePartition(q_tot, FT(0), q_tot)) # Assume all ice
+    T, converged = find_zero(
+      T -> internal_energy_sat(T, ρ, q_tot) - e_int,
+      T_ -> roots_∂e_int_∂T(T_, e_int, ρ, q_tot),
+      T_1,
+      NewtonsMethod(), FT(1e-3), 10)
+      # NewtonsMethodAD(), FT(1e-3), 3)
```

Comment:
Limiting to ~3 iterations should suffice once the gradients are right.

Reply:
3 iterations seem to work really well for low values of `q_tot`, but the error is larger at high values. If we print:
```
  q_tot = FT(0); ρ = FT(1)
  @show abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0)

  q_tot = FT(0.21); ρ = FT(0.1)
  @show abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(200.0, ρ, q_tot), ρ, q_tot) - 200.0)

  q_tot = FT(0.78); ρ = FT(1)
  @show abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0)
```

We get

## 3 iterations
```
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0) = 0.0
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(200.0, ρ, q_tot), ρ, q_tot) - 200.0) = 1.9237518245063256e-9
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0) = 113.0644848960498 # not converged
```

## 10 iterations

```
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0) = 0.0
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(200.0, ρ, q_tot), ρ, q_tot) - 200.0) = 1.9237518245063256e-9
abs(saturation_adjustment_NewtonsMethod(internal_energy_sat(300.0, ρ, q_tot), ρ, q_tot) - 300.0) = 0.004837695608841841 # not converged
```

This seems to be converging to the correct solution given enough iterations quickly at low `q_tot`. I'd suggest leaving `maxiter` and the tolerance as is for now, and we can search for optimal parameters and compare all of the methods systematically in another PR.

Reply:
Ok. `q_tot > 0.1` will never occur on Earth (and if such high `q_tot` occur on other planets, saturation adjustment is a bad idea anyway). I think it would be good to add an option of just allowing one iteration, and make that the default. For what we need in LES, that will do, and it avoids the performance problems arising from iterations.

Reply:
Sounds good to me. I'll remove that test and reduce the iterations to 3.